x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.test.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { shallow, ShallowWrapper } from 'enzyme';
+
+import { EuiComboBox, EuiFieldText } from '@elastic/eui';
+
+import { AttributeSelector } from './attribute_selector';
+import { ANY_AUTH_PROVIDER, ANY_AUTH_PROVIDER_OPTION_LABEL } from './constants';
+
+const handleAttributeSelectorChange = jest.fn();
+const handleAttributeValueChange = jest.fn();
+const handleAuthProviderChange = jest.fn();
+
+const baseProps = {
+  attributeName: 'An Attribute',
+  attributeValue: 'Something',
+  attributes: ['a', 'b', 'c'],
+  availableAuthProviders: ['ees_saml', 'kbn_saml'],
+  selectedAuthProviders: ['ees_saml'],
+  elasticsearchRoles: ['whatever'],
+  multipleAuthProvidersConfig: true,
+  disabled: false,
+  handleAttributeSelectorChange,
+  handleAttributeValueChange,
+  handleAuthProviderChange,
+};
+
+describe('AttributeSelector', () => {
+  it('renders', () => {
+    const wrapper = shallow(<AttributeSelector {...baseProps} />);
+
+    expect(wrapper.find('[data-test-subj="attributeSelector"]')).toBeDefined();
+  });
+
+  it('renders disabled panel with className', () => {
+    const wrapper = shallow(<AttributeSelector {...baseProps} disabled />);
+
+    expect(wrapper.find('[data-test-subj="attributeSelector"]').prop('className')).toEqual(
+      'euiPanel--disabled'
+    );
+  });
+
+  describe('Auth Providers', () => {
+    const findAuthProvidersSelect = (wrapper: ShallowWrapper) =>
+      wrapper.find('[data-test-subj="authProviderSelect"]');
+
+    it('will not render if "availableAuthProviders" prop has not been provided', () => {
+      const wrapper = shallow(
+        <AttributeSelector {...baseProps} availableAuthProviders={undefined} />
+      );
+
+      expect(wrapper.find('[data-test-subj="authProviderSelect"]')).toHaveLength(0);
+    });
+
+    it('handles fallback props', () => {
+      const wrapper = shallow(
+        <AttributeSelector
+          {...baseProps}
+          attributeValue={undefined}
+          selectedAuthProviders={undefined}
+        />
+      );
+
+      const select = findAuthProvidersSelect(wrapper) as any;
+
+      expect(select.prop('selectedOptions')).toEqual([
+        {
+          label: ANY_AUTH_PROVIDER_OPTION_LABEL,
```

Comment:
I noticed here you use `ANY_AUTH_PROVIDER_OPTION_LABEL` and below you test on just the text `Any`. You should probably do it consistently in both places, and I was actually thinking if you just wrote the text `Any` here as well it may help make this test more understandable?

Reply:
I chose to use the constant in case it ever changes. Modern IDEs can reveal the value quite easily 
![image](https://user-images.githubusercontent.com/1869731/108546210-6cca3800-72ae-11eb-9a71-f58f8e077cf7.png)

Reply:
Cool, that's fine. It's a trade-off for sure, I definitely prefer string literals just so that the test is human readable which helps to understand what the component actually does. Thanks for making the change.